src/Microsoft.TemplateEngine.Abstractions/ISettingsLoader.cs

```diff
@@ -28,7 +28,7 @@ public interface ISettingsLoader
 
         bool TryGetMountPointInfo(Guid mountPointId, out MountPointInfo info);
 
-        void WriteTemplateCache(IList<ITemplateInfo> templates, string locale, bool isCurrentCache);
+        void WriteTemplateCache(IList<ITemplateInfo> templates, string cacheVersion, string locale, bool isCurrentCache);
```

Comment:
Thinking more about this - I don't think the version stamp for the cache belongs on the template cache - getting the feeling that this would be more at home on the settings file itself (especially since the template cache & settings file must be coordinated anyway)